### PR TITLE
Fix release-notes make target to work with new process

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Follow up on #4366, this fixes the make target for generating release notes to use commit SHAs instead of tags since the tag will no yet exist when CAPZ devs run it locally (the tag will be created later by the github action). I used these changes to generate the notes in #4369.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by willie-yao — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Comment by CecileRobertMichon** [comment]
/hold

I just saw something that needs to be fixed

**Comment by CecileRobertMichon** [comment]
Done 

@mboersma I kept the commits unsquashed for easier review of what changed, let me know when I should squash

**Review by mboersma — COMMENTED** [review]
@CecileRobertMichon I see your new change. Still lgtm, so squash away.

**Comment by CecileRobertMichon** [comment]
Let's merge https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4377 first to make sure every works correctly

**Comment by CecileRobertMichon** [comment]
Rebased + squashed and removed the old release notes file

**Review by mboersma — APPROVED** [review]
/lgtm

File: docs/book/src/developers/releasing.md [hunk 1]
```diff
@@ -73,11 +73,19 @@ This can be done in parallel with release publishing and does not impact the rel
...
+
+    ```sh
+    git fetch upstream
+    git checkout main
+    ```
 
 1. Generate release notes by running the following command:
 
     ```sh
+    export GITHUB_TOKEN=<your GH token>
```
Comment (CecileRobertMichon) [inline docs/book/src/developers/releasing.md:88]: added this

**Review by CecileRobertMichon — COMMENTED** [review]
this is ready for real now

/hold cancel

**Pull request merged**